### PR TITLE
fixing safety inference and safety adapter for new API spec. Pinned t…

### PR DESCRIPTION
**Pull request by yogishbaliga**
…he llama_models version to 0.0.24 as the latest version 0.0.35 has the model descriptor name changed. I was getting the missing package error during runtime as well, hence added the dependency to requirements.txt

This was tested with local instance of llama-stack in container as well as without container

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ blobfile
 fire
 httpx
 huggingface-hub
-llama-models>=0.0.24
+llama-models==0.0.24
```
Comment (ashwinb) [inline requirements.txt:5]: I think this should be >=0.0.36 and cannot be changed. What issue does this cause?

**Review by ashwinb — CHANGES_REQUESTED** [review]
Everything looks great other than the change in requirements.txt

File: requirements.txt (hunk 1 above)
Comment (yogishbaliga) [reply requirements.txt]: It was causing the naming of the models in inference adapter. This is feed now.

**Comment by yogishbaliga** [comment]
> Everything looks great other than the change in requirements.txt

@ashwinb requirements.txt is fixed and has no changes anymore.

File: llama_stack/providers/adapters/inference/together/config.py [hunk 2]
```diff
@@ -4,17 +4,12 @@
...
 
-from pydantic import BaseModel, Field
-
 from llama_models.schema_utils import json_schema_type
-
-from llama_stack.distribution.request_headers import annotate_header
+from pydantic import BaseModel, Field
 
 
 class TogetherHeaderExtractor(BaseModel):
```
Comment (ashwinb) [inline llama_stack/providers/adapters/inference/together/config.py:11]: is this class needed? I think you only need the ProviderDataValidator class right?

File: llama_stack/providers/adapters/safety/together/together.py [hunk 3]
```diff
@@ -3,12 +3,40 @@
...
+    SafetyViolation,
+    ViolationLevel,
+)
 from llama_stack.distribution.request_headers import get_request_provider_data
 
-from .config import TogetherProviderDataValidator, TogetherSafetyConfig
+from .config import TogetherSafetyConfig
+
+SAFETY_SHIELD_TYPES = {
+    "Llama-Guard-3-8B": "meta-llama/Meta-Llama-Guard-3-8B",
```
Comment (ashwinb) [inline llama_stack/providers/adapters/safety/together/together.py:21]: does the vision shield (Llama-Guard-3-11B-Vision) also work? i.e., is the model deployed on the backend? If so I will make it work with the rest of the infra on the Stack.

**Review by ashwinb — APPROVED** [review]
Looks good! I'd still ask for the Test Plan to be expressed in a lot more detail -- e.g., how you built the stack (exact commands), how you tested everything end-to-end, etc.

File: llama_stack/providers/adapters/safety/together/together.py (hunk 3 above)
Comment (yogishbaliga) [reply llama_stack/providers/adapters/safety/together/together.py:21]: vision shield is added here

File: llama_stack/providers/adapters/safety/together/together.py [hunk 4]
```diff
@@ -21,48 +50,42 @@ async def initialize(self) -> None:
...
-        messages=messages, model="meta-llama/Meta-Llama-Guard-3-8B"
-    )
+    response = client.chat.completions.create(messages=messages, model=model_name)
     if len(response.choices) == 0:
         return None
 
     response_text = response.choices[0].message.content
     if response_text == "safe":
-        return None
+        return SafetyViolation(
```
Comment (ashwinb) [inline llama_stack/providers/adapters/safety/together/together.py:86]: this is not right! if there is no violation you should not return a Violation object?

**Review by ashwinb — CHANGES_REQUESTED** [review]

**Pull request merged**